File: src/Github-Analyzer.WebApp/src/types/analysis/statistic-analysis.ts
// Hasil analisis statistik dari backend StatisticAnalysis entity
export interface StatisticAnalysis {
  id: string
  projectId: string
  branch: string | null
  commitHash: string | null
  generatedAtUtc: string | null

  // Structural
  totalFolders: number | null
  totalFiles: number | null
  sizeInBytes: number | null

  // Code lines
  totalLinesOfCode: number | null
  codeLines: number | null
  commentLines: number | null
  blankLines: number | null

  // Git (from GitHub API — may be null if API unreachable)
  totalCommits: number | null
  totalContributors: number | null
  totalBranches: number | null
}

File: src/Github-Analyzer.WebApi/Entities/Analysis/StatisticAnalysis.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using GithubAnalyzer.WebApi.Entities.Auth;
using GithubAnalyzer.WebApi.Entities.Repo;

namespace GithubAnalyzer.WebApi.Entities.Analysis;

[Table("StatisticAnalyses", Schema = "Repo")]
public class StatisticAnalysis : BaseEntity
{
    [Required] public Guid UserId { get; set; }
    [Required] public Guid ProjectId { get; set; }

    [ForeignKey(nameof(UserId))] 
    public ApplicationUser User { get; set; } = default!;
    
    [ForeignKey(nameof(ProjectId))]
    public Project Project { get; set; } = default!;

    [MaxLength(64)] public string? Branch { get; set; }
    [MaxLength(64)] public string? CommitHash { get; set; }
    
    public DateTime? GeneratedAtUtc { get; set; }

    // Structural Statistics
    public int? TotalFolders { get; set; }
    public int? TotalFiles { get; set; }
    public int? SizeInBytes { get; set; }

    // Code Statistics
    public long? TotalLinesOfCode { get; set; }
    public long? CodeLines { get; set; }
    public long? CommentLines { get; set; }
    public long? BlankLines { get; set; }

    // Git Statistics
    public int? TotalCommits { get; set; }
    public int? TotalContributors { get; set; }
    public int? TotalBranches { get; set; }
    
    [Required, MaxLength(50)]
    public string AnalysisVersion { get; set; } = default!;
}
